packages/utils/src/utils/index.js

```diff
@@ -182,13 +182,38 @@ export function formatLabels(labelsRaw) {
   return formattedLabelsToRender;
 }
 
+// Sorts the steps by finishedAt and startedAt timestamps
+export function sortByTimestamp(steps) {
+  return steps.sort((i, j) => {
+    const iFinishedAt = Date.parse(i.stepStatus?.terminated?.finishedAt);
+    const jFinishedAt = Date.parse(j.stepStatus?.terminated?.finishedAt);
+    const iStartedAt = Date.parse(i.stepStatus?.terminated?.startedAt);
+    const jStartedAt = Date.parse(j.stepStatus?.terminated?.startedAt);
+
+    if (!iFinishedAt || !jFinishedAt) {
+      return 0;
+    }
+    if (iFinishedAt !== jFinishedAt) {
+      return iFinishedAt - jFinishedAt;
+    }
+    // if finishedAt timestamps are equal, compare the startedAt timestamps
+    if (!iStartedAt || !jStartedAt) {
+      return 0;
+    }
+    if (iStartedAt !== jStartedAt) {
```

Comment:
should be able to `return iStartedAt - jStartedAt;` here and avoid the additional `if` and `return 0`